torch/_dynamo/codegen.py

```diff
@@ -590,22 +590,26 @@ def make_call_generated_code(self, fn_name: str) -> None:
         seen_sources: OrderedSet[Source] = OrderedSet()
 
         def collect_temp_source(source):
-            if source in seen_sources:
-                # This source is used atleast twice, so it can be reused
-                self.mark_source_temp(source)
-                # Dont trace source further. This prevents us from marking too
-                # many nodes as temp sources.
-                return
-
-            seen_sources.add(source)
-
-            if isinstance(source, ChainedSource):
-                collect_temp_source(source.base)
-
-            if isinstance(source, DictGetItemSource) and isinstance(
-                source.index, Source
-            ):
-                collect_temp_source(source.index)
+            stack = collections.deque([source])
+            while stack:
+                source = stack.pop()
+
+                if source in seen_sources:
+                    # This source is used at least twice, so it can be reused
+                    self.mark_source_temp(source)
+                    # Dont trace source further. This prevents us from marking too
+                    # many nodes as temp sources.
+                    return
```

Comment:
This would end the search? Should we be `continue`ing?

Reply:
Per ezyang suggestion, instead of fixing the algorithms, it is preferred to explicitly break the connections to the cycles on cleanup for Dynamo. So this change is no longer needed anymore. Please review the latest method.